src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -982,6 +982,257 @@ private Task WriteSuffix()
             Output.ProducingComplete(end);
         }
 
+        public bool TakeStartLineSpan(ReadableBuffer buffer, out ReadCursor consumed, out ReadCursor examined)
+        {
+            var start = buffer.Start;
+            var end = buffer.Start;
+
+            examined = buffer.End;
+            consumed = buffer.Start;
+
+            if (_requestProcessingStatus == RequestProcessingStatus.RequestPending)
+            {
+                ConnectionControl.ResetTimeout(_requestHeadersTimeoutMilliseconds, TimeoutAction.SendTimeoutResponse);
+            }
+
+            _requestProcessingStatus = RequestProcessingStatus.RequestStarted;
+
+            ReadableBuffer limitedBuffer;
+            if (buffer.Length >= ServerOptions.Limits.MaxRequestLineSize)
+            {
+                limitedBuffer = buffer.Slice(0, ServerOptions.Limits.MaxRequestLineSize);
+            }
+            else
+            {
+                limitedBuffer = buffer;
+            }
+
+            if (ReadCursorOperations.Seek(limitedBuffer.Start, limitedBuffer.End, out end, ByteLF) == -1)
+            {
+                if (limitedBuffer.Length == ServerOptions.Limits.MaxRequestLineSize)
+                {
+                    RejectRequest(RequestRejectionReason.RequestLineTooLong);
+                }
+                else
+                {
+                    return false;
+                }
+            }
+
+            const int stackAllocLimit = 512;
+
+            // Move 1 byte past the \r
+            end = limitedBuffer.Move(end, 1);
+            var startLineBuffer = limitedBuffer.Slice(0, end);
+
+            Span<byte> span;
+
+            if (startLineBuffer.IsSingleSpan)
```

Comment:
Still the same issue with this pattern.. Add a buffer to Frame and reuse it for all parsers.